src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -39,7 +39,7 @@ public CompileContext(ProjectContext rootProject, BuilderCommandApp args)
             _args = (BuilderCommandApp) args.ShallowCopy();
 
             // Set up Output Paths. They are unique per each CompileContext
-            _args.OutputValue = _rootProject.GetOutputPath(_args.ConfigValue, _args.OutputValue);
+            _args.OutputValue = _rootProject.GetOutputPathCalculator(_args.OutputValue).BaseRootOutputPath;
             _args.IntermediateValue = _rootProject.GetIntermediateOutputPath(_args.ConfigValue, _args.IntermediateValue, _args.OutputValue);
```

Comment:
For consistency I would make the output calculator also handle the intermediary path. It looks a bit weird here when a client of the API has to call two different things to get the output paths.